diff --git a/travis/install-autotools.sh b/travis/install-autotools.sh
index b1c5160..cd9fa05 100755
--- a/travis/install-autotools.sh
+++ b/travis/install-autotools.sh
@@ -109,7 +109,8 @@ else
     if [ -f secure_snprintf.patch ] ; then
         echo secure_snprintf.patch already exists! Using existing copy.
     else
-        ${download} secure_snprintf.patch https://raw.githubusercontent.com/macports/macports-ports/master/devel/m4/files/secure_snprintf.patch
+        pwd
+        ${download} secure_snprintf.patch https://raw.githubusercontent.com/macports/macports-ports/v2.7.0-archive/devel/m4/files/secure_snprintf.patch
     fi
     if patch -p0 -N < secure_snprintf.patch ; then
         echo patch applied
